pkg/kritis/cron/cron.go

```diff
@@ -91,22 +89,16 @@ func Start(ctx context.Context, cfg Config, checkInterval time.Duration) {
 
 // CheckPods checks all running pods against defined policies.
 func CheckPods(cfg Config, isps []v1beta1.ImageSecurityPolicy) error {
+	r := review.New(cfg.Client, cfg.ViolationStrategy, cfg.ViolationChecker)
 	for _, isp := range isps {
 		ps, err := cfg.PodLister(isp.Namespace)
 		if err != nil {
 			return err
 		}
 		for _, p := range ps {
-			for _, c := range pods.Images(p) {
-				v, err := cfg.ViolationChecker(c, isp)
-				if err != nil {
-					return err
-				}
-				if len(v) != 0 {
-					if err := cfg.ViolationStrategy.HandleViolation(c, &p, v); err != nil {
-						glog.Errorf("handling violations: %s", err)
-					}
-				}
+			glog.Infof("Checking po %s", p.Name)
+			if err := r.Review(pods.Images(p), isps, &p); err != nil {
+				glog.Error(err)
```

Comment:
This doesn't return an error: how does the CheckPods() caller know there was a violation?

Reply:
CheckPods returns an error if there was error checking a pod. It does not care about violations if found.  The violationStrategy is supposed to handle it. 
Does that make sense?

Reply:
Makes sense, thanks.